api/src/test/java/org/openmrs/api/EncounterServiceTest.java

```diff
@@ -382,6 +384,20 @@ public void saveEncounter_shouldSaveEncounterWithComplexObs() {
 		assertEquals(2, encounter.getObsAtTopLevel(true).size());
 	}
 	
+	@Test
+	public void saveEncounter_shouldSaveEncounterWithCondition() {
+		Encounter encounter = buildEncounter();
+		Condition condition = new Condition();
+		
+		condition.setPatient(encounter.getPatient());
+		encounter.setConditions(Collections.singleton(condition));
+		EncounterService es = Context.getEncounterService();
+		es.saveEncounter(encounter);
+		
+		encounter = es.getEncounter(encounter.getId());
+		assertEquals(1, encounter.getConditions().size());
```

Comment:
We need to assert that the conditions that are returned are actually those that are expected. Use two conditions as well to enforce the use of an actual "_collection_".

Reply:
Always use the given/when/then pattern when writing tests. This makes easier for the reader to find what you actually replay.

Unless @dkayiwa thinks differently?

Reply:
I like conventions that make code easier to understand. 👍